### PR TITLE
Feature/UNR-3291

### PR DESCRIPTION
**Pull request by yujunting**
#### Description
Change icon of 'start' toolbar button based on selected connection flow

#### Release note
when choosing the options in 'native/local/cloud'  of  'Start' toolbar button, change the start button's icon

#### Tests
Only editor code change. So that test it in the editor. Click 'start' toolbar button when choosing a different option such as 'Do not connect/local depolyment/cloud deployment'

#### Documentation
None

image
image
image
image

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Public/SpatialGDKEditorToolbarStyle.h [hunk 1]
```diff
@@ -14,8 +14,7 @@ class FSpatialGDKEditorToolbarStyle
 	static void Shutdown();
 	static void ReloadTextures();
 	static const ISlateStyle& Get();
-	static FName GetStyleSetName();
-
+	static FName GetStyleSetName();
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Public/SpatialGDKEditorToolbarStyle.h:17]:
```suggestion
	static FName GetStyleSetName();

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -76,6 +76,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Added `OnPlayerSpawnFailed` delegate to `SpatialGameInstance`. This is helpful if you have established a successful connection but the server worker crashed. `OnConnected` and `OnConnectionFailed` are now also blueprint-assignable.
 - `Launch Deployment` in the `Cloud Deployment` dialog will now generate schema, generate a snapshot, build the server worker, build the client, build the simulate player worker, and upload the assembly before launching the deployment all in one step, there are checkboxes to toggle generation of schema and snapshots as well as whether to build the client worker.
 - You can now edit the project name field in the `Cloud Deployment` window.
+- Change icon of 'start' toolbar button based on selected connection flow
```
Comment (improbable-valentyn) [inline CHANGELOG.md:79]:
```suggestion
- Change the icon of the 'Start' toolbar button based on the selected connection flow.
```

**Comment by improbable-valentyn** [comment]
We definitely shouldn't have a button that does nothing, that's just confusing. I'm also not sure if the behavior would be intuitive enough if it's just different based on the icon. I suggest that we also change the text below the icon to "Local Deployment", "Cloud Deployment", and "No Deployment", and grey out the button for "No Deployment". @jessicafalk @jayprobable @raymonwang thoughts?

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 3]
```diff
@@ -799,11 +817,39 @@ void FSpatialGDKEditorToolbarModule::VerifyAndStartDeployment()
...
+
+void FSpatialGDKEditorToolbarModule::StartLocalSpatialDeploymentButtonClicked()
 {
 	VerifyAndStartDeployment();
 }
 
+void FSpatialGDKEditorToolbarModule::StartCloudSpatialDeploymentButtonClicked()
+{
+	// Create and open the cloud configuration dialog
+	SimulatedPlayerDeploymentWindowPtr = SNew(SWindow)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:833]: This code should be moved into a function that's called both here and from `ShowSimulatedPlayerDeploymentDialog`. Also, we should probably update the naming of this window from "SimulatedPlayerDeployment" to just "CloudDeployment" (the original feature was focused around simulated players but this is no longer the case).

Comment (yujunting) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: I use ShowSimulatedPlayerDeploymentDialog() to replace it

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarStyle.cpp [hunk 4]
```diff
@@ -61,11 +61,23 @@ TSharedRef<FSlateStyleSet> FSpatialGDKEditorToolbarStyle::Create()
 	Style->Set("SpatialGDKEditorToolbar.CreateSpatialGDKSchema.Small",
 		new IMAGE_BRUSH(TEXT("Schema@0.5x"), Icon20x20));
 
-	Style->Set("SpatialGDKEditorToolbar.StartSpatialDeployment",
+	Style->Set("SpatialGDKEditorToolbar.StartNativeUnrealDeployment",
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarStyle.cpp:64]:
```suggestion
	Style->Set("SpatialGDKEditorToolbar.StartNoAutomaticConnection",
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarStyle.cpp [hunk 5]
```diff
@@ -61,11 +61,23 @@ TSharedRef<FSlateStyleSet> FSpatialGDKEditorToolbarStyle::Create()
...
 		new IMAGE_BRUSH(TEXT("Launch"), Icon40x40));
 
-	Style->Set("SpatialGDKEditorToolbar.StartSpatialDeployment.Small",
+	Style->Set("SpatialGDKEditorToolbar.StartNativeUnrealDeployment.Small",
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarStyle.cpp:67]:
```suggestion
	Style->Set("SpatialGDKEditorToolbar.StartNoAutomaticConnection.Small",
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 6]
```diff
@@ -835,7 +863,22 @@ void FSpatialGDKEditorToolbarModule::LaunchInspectorWebpageButtonClicked()
...
 
-bool FSpatialGDKEditorToolbarModule::StartSpatialDeploymentIsVisible() const
+bool FSpatialGDKEditorToolbarModule::StartNoAutomaticConnectionIsVisible() const
+{
+	return GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking() && GetDefault<USpatialGDKEditorSettings>()->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
+}
+
+bool FSpatialGDKEditorToolbarModule::StartNoAutomaticConnectionCanExecute() const
+{
+	return GetDefault<USpatialGDKEditorSettings>()->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:873]: I think this should be always greyed out.
```suggestion
	return false;
```

Comment (yujunting) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: it makes sense that disabled the button if use NoAutomaticConnection

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp [hunk 7]
```diff
@@ -10,7 +10,9 @@ void FSpatialGDKEditorToolbarCommands::RegisterCommands()
 	UI_COMMAND(CreateSpatialGDKSchemaFull, "Schema (Full Scan)", "Creates SpatialOS Unreal GDK schema for all assets.", EUserInterfaceActionType::Button, FInputGesture());
 	UI_COMMAND(DeleteSchemaDatabase, "Delete schema database", "Deletes the scheme database file", EUserInterfaceActionType::Button, FInputGesture());
 	UI_COMMAND(CreateSpatialGDKSnapshot, "Snapshot", "Creates SpatialOS Unreal GDK snapshot.", EUserInterfaceActionType::Button, FInputGesture());
-	UI_COMMAND(StartSpatialDeployment, "Start", "Starts a local instance of SpatialOS.", EUserInterfaceActionType::Button, FInputGesture());
+	UI_COMMAND(StartNoAutomaticConnection, "Start", "Not Automatic Connect to service", EUserInterfaceActionType::Button, FInputGesture());
```
Comment (jayprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp:13]:
```suggestion
	UI_COMMAND(StartNoAutomaticConnection, "None", "Not Automatic Connect to service", EUserInterfaceActionType::Button, FInputGesture());
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp [hunk 8]
```diff
@@ -10,7 +10,9 @@ void FSpatialGDKEditorToolbarCommands::RegisterCommands()
...
+	UI_COMMAND(StartLocalSpatialDeployment, "Start", "Connect to local deployment service", EUserInterfaceActionType::Button, FInputGesture());
```
Comment (jayprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp:14]:
```suggestion
	UI_COMMAND(StartLocalSpatialDeployment, "Local", "Connect to local deployment service", EUserInterfaceActionType::Button, FInputGesture());
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp [hunk 9]
```diff
@@ -10,7 +10,9 @@ void FSpatialGDKEditorToolbarCommands::RegisterCommands()
...
+	UI_COMMAND(StartCloudSpatialDeployment, "Start", "Connect to cloud SpatialOS service", EUserInterfaceActionType::Button, FInputGesture());
```
Comment (jayprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbarCommands.cpp:15]:
```suggestion
	UI_COMMAND(StartCloudSpatialDeployment, "Cloud", "Connect to cloud SpatialOS service", EUserInterfaceActionType::Button, FInputGesture());
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 10]
```diff
@@ -835,7 +863,23 @@ void FSpatialGDKEditorToolbarModule::LaunchInspectorWebpageButtonClicked()
 	}
 }
 
-bool FSpatialGDKEditorToolbarModule::StartSpatialDeploymentIsVisible() const
+bool FSpatialGDKEditorToolbarModule::StartNoAutomaticConnectionIsVisible() const
+{
+	auto settings = GetDefault<UGeneralProjectSettings>();
+	return (false == settings->UsesSpatialNetworking()) || GetDefault<USpatialGDKEditorSettings>()->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:869]:
```suggestion
	return (!GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking()) || GetDefault<USpatialGDKEditorSettings>()->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
```

**Pull request merged**